react-app/src/App.tsx

```diff
@@ -0,0 +1,55 @@
+import {Button} from '@material-ui/core';
+import * as React from 'react';
+import {useEffect, useState} from 'react';
+import './App.css';
+import TutorialOverview from './components/TutorialOverview';
+import {VSCodeAPI} from './VSCodeAPI';
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export default function App() {
+
+  const [tutorials, setTutorials] = useState<Array<any>>([]);
+  const [selectedTutorial, selectTutorial] = useState(-1);
+
+  useEffect(() => {
+    return VSCodeAPI.onMessage((message) => {
+
+      switch (message.data.command) {
+        case 'setTutorials':
+          setTutorials(message.data.tutorials);
+          break;
+      }
+    });
+  });
+
+  function createTutorialList() {
```

Comment:
I would suggest to use `const createTutorialList = () => {...}` format instead. This way you won't get problems with `this` as the scope stays correct.